src/main/java/io/appium/java_client/service/local/AppiumDriverLocalService.java

```diff
@@ -294,6 +294,37 @@ public void addOutPutStreams(List<OutputStream> outputStreams) {
         }
     }
 
+    /**
+     * Remove the outputStream which is receiving server output data
+     *
+     * @return true if the outputStream has been removed
+     */
+    public boolean removeOutputStream(OutputStream outputStream) {
+        checkNotNull(outputStream, "outputStream parameter is NULL!");
+        return stream.remove(outputStream);
+    }
+
+    /**
+     * Remove the outputStream which is receiving server output data, and close
+     *
+     * @param outputStream
+     * @return true if the outputStream has been removed
+     */
+    public boolean removeOutputStreamAndClose(OutputStream outputStream) throws IOException {
+        boolean removed = removeOutputStream(outputStream);
```

Comment:
I would rather return the removed stream as Optional, so we could close it in the client code if needed. So these two helpers would not be really necessary